src/providers/virtual/qgsvirtuallayerprovider.cpp

```diff
@@ -555,11 +557,16 @@ QgsRectangle QgsVirtualLayerProvider::extent() const
 void QgsVirtualLayerProvider::updateStatistics() const
 {
   bool hasGeometry = mDefinition.geometryWkbType() != QgsWkbTypes::NoGeometry;
-  QString subset = mSubset.isEmpty() ? QString() : " WHERE " + mSubset;
+  QString subset = mSubset.isEmpty() ? QString() : QStringLiteral( " WHERE %1" ).arg( mSubset );
+
+  Q_ASSERT( ! mTableName.isNull() );
```

Comment:
`mTableName` might be an empty string when the layer is broken, because `createIt` is never called. This ASSERT should catch situations like the one that this PR is targeting.